src/manifolds/SegreWarped.jl

```diff
@@ -0,0 +1,284 @@
+@doc raw"""
+    WarpedMetric{A} <: AbstractMetric
+
+is the ``A``-warped metric on the Segre manifold ``\mathcal{S}``. We denote it ``\mathcal{S}_A``. It is a generalization of the metric
+````math
+    \langle (\nu, u_1, \dots, u_d), (\xi, v_1, \dots, v_d) \rangle_{(\lambda, x_1, \dots, x_d)} = \nu \xi + \lambda^2 (\langle u_1, v_1 \rangle + \dots + \langle u_d, v_d \rangle),
+````
+on the Segre that is induced from its Euclidean embedding, to the metric
+````math
+    \langle (\nu, u_1, \dots, u_d), (\xi, v_1, \dots, v_d) \rangle_{(\lambda, x_1, \dots, x_d)} = \nu \xi + (A \lambda)^2 (\langle u_1, v_1 \rangle + \dots + \langle u_d, v_d \rangle).
+````
+``A`` is called the _warping factor_ and ``A = 1`` corresponds to the usual Segre manifold.
+
+The geometry is summarized in [JacobssonSwijsenVandervekenVannieuwenhoven:2024](@cite).
+"""
+struct WarpedMetric{A} <: AbstractMetric end
+
+valence(::MetricManifold{𝔽,Segre{𝔽,V},WarpedMetric{A}}) where {V,A,𝔽} = V
+ndims(::MetricManifold{𝔽,Segre{𝔽,V},WarpedMetric{A}}) where {V,A,𝔽} = length(V)
```

Comment:
See comment in the Segre manifold – are these needed – should they be documented/exported?